src/components/phoneViewport.vue

```diff
@@ -0,0 +1,458 @@
+<template>
+  <div class="docs-demo-mode-toggle">
+    <button
+      :class="iosSelected"
+      @click="toggleIosSelected"
+      title="Toggle iOS mode"
+    >
+      iOS</button
+    ><button
+      :class="androidSelected"
+      @click="toggleAndroidSelected"
+      title="Toggle Android mode"
+    >
+      Android
+    </button>
+  </div>
+  <div :class="divClass">
+    <figure>
+      <svg class="docs-demo-device__md-bar" viewBox="0 0 1384.3 40.3">
+        <path
+          class="st0"
+          d="M1343 5l18.8 32.3c.8 1.3 2.7 1.3 3.5 0L1384 5c.8-1.3-.2-3-1.7-3h-37.6c-1.5 0-2.5 1.7-1.7 3z"
+        ></path>
+        <circle class="st0" cx="1299" cy="20.2" r="20"></circle>
+        <path
+          class="st0"
+          d="M1213 1.2h30c2.2 0 4 1.8 4 4v30c0 2.2-1.8 4-4 4h-30c-2.2 0-4-1.8-4-4v-30c0-2.3 1.8-4 4-4zM16 4.2h64c8.8 0 16 7.2 16 16s-7.2 16-16 16H16c-8.8 0-16-7.2-16-16s7.2-16 16-16z"
+        ></path></svg
+      ><svg class="docs-demo-device__ios-notch" viewBox="0 0 219 31">
+        <path
+          d="M0 1V0h219v1a5 5 0 0 0-5 5v3c0 12.15-9.85 22-22 22H27C14.85 31 5 21.15 5 9V6a5 5 0 0 0-5-5z"
+          fill-rule="evenodd"
+        ></path></svg
+      ><iframe :src="frameSrc">
+        <RouterView />
+      </iframe>
+    </figure>
+  </div>
+</template>
+
+<script>
+import { ref } from 'vue'
+
+export default {
+  name: 'phoneViewport',
+  setup() {
+    const iosSelected = ref('is-selected')
+    const androidSelected = ref('')
+    const divClass = ref('docs-demo-device ios')
+    const frameSrc = ref('http://localhost:8080?ionic:mode=ios')
```

Comment:
should this work when deployed to a public server?

Reply:
def not, we should have the needed info in the router via `useRouter` if I'm not mistaken

Reply:
the mode could also be a config of ionic/vue, although I'm not sure if a ref would be supported there